cli/cycpp.py

```diff
@@ -1380,19 +1411,34 @@ def impl(self, ind="  "):
             if not isinstance(info, Mapping):
                 # this member is a variable alias pointer
                 continue
-
             if self.pragmaname in info:
                 impl += info[self.pragmaname]
                 continue
+
+            # get expression
             t = info["type"]
+            key = t if isinstance(t, STRING_TYPES) else t[0]
             if t in BUFFERS:
```

Comment:
Should this be `if key in BUFFERS:`? Seems to be trying to do something similar to line 1305.